### PR TITLE
ItemLists now have a scrollbar and can be scrolled with the mouse wheel as well.

### PR DESCRIPTION
**Pull request by gradientvera**

### CODE REVIEW

File: Robust.Client/UserInterface/Controls/ItemList.cs [hunk 1]
```diff
@@ -1,19 +1,29 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using System.Drawing;
```
Comment (PJB3005) [inline Robust.Client/UserInterface/Controls/ItemList.cs]: What did you include `System.Drawing` for here?

File: Robust.Client/UserInterface/Controls/ItemList.cs [hunk 2]
```diff
@@ -56,6 +70,10 @@ public void AddItem(string text, Texture icon = null, bool selectable = true)
             {
                 var i = new Item() {Text = text, Icon = icon, Selectable = selectable};
                 _itemList.Add(i);
+                if (_isAtBottom && ScrollFollowing)
```
Comment (PJB3005) [inline Robust.Client/UserInterface/Controls/ItemList.cs:86]: You're not re-calculating `_totalContentHeight` here before running this so this won't work correctly.

File: Robust.Client/UserInterface/Controls/ItemList.cs [hunk 3]
```diff
@@ -426,7 +451,22 @@ protected internal override void Draw(DrawingHandleScreen handle)
                 }
 
                 separation += (0, itemHeight);
+                _itemListHeight += itemHeight;
+            }
+
+            _totalContentHeight = (int)_itemListHeight;
```
Comment (PJB3005) [inline Robust.Client/UserInterface/Controls/ItemList.cs]: Setting this inside the drawing loop is a bad idea.

File: Robust.Client/UserInterface/Controls/ItemList.cs [hunk 4]
```diff
@@ -453,6 +493,56 @@ protected void DrawTextInternal(DrawingHandleScreen handle, string text, UIBox2
             }
         }
 
+        protected override Vector2 CalculateMinimumSize()
+        {
+            return (ActualBackground?.MinimumSize.X ?? 0, _itemListHeight);
```
Comment (PJB3005) [inline Robust.Client/UserInterface/Controls/ItemList.cs]: You need to accommodate for the the horizontal size of the background too.

**Review by PJB3005 — CHANGES_REQUESTED** [review]

File: Robust.Client/UserInterface/Controls/ItemList.cs (hunk 1 above)
Comment (gradientvera) [reply Robust.Client/UserInterface/Controls/ItemList.cs]: Honestly, I'm not even sure myself. Rider must have screwed something up.
I'll remove it

File: Robust.Client/UserInterface/Controls/ItemList.cs (hunk 4 above)
Comment (gradientvera) [reply Robust.Client/UserInterface/Controls/ItemList.cs]: Hmm, I'm not too sure what you mean here... Isn't MinimumSize.X enough, or should I do something else?

File: Robust.Client/UserInterface/Controls/ItemList.cs (hunk 3 above)
Comment (gradientvera) [reply Robust.Client/UserInterface/Controls/ItemList.cs]: Should be fixed now.

File: Robust.Client/UserInterface/Controls/ItemList.cs (hunk 2 above)
Comment (gradientvera) [reply Robust.Client/UserInterface/Controls/ItemList.cs:86]: Should be fixed now.

File: Robust.Client/UserInterface/Controls/ItemList.cs (hunk 4 above)
Comment (PJB3005) [reply Robust.Client/UserInterface/Controls/ItemList.cs]: I meant vertical size, durr.

Comment (gradientvera) [reply Robust.Client/UserInterface/Controls/ItemList.cs]: Should be good now, then

File: Robust.Client/UserInterface/Controls/ItemList.cs [hunk 5]
```diff
@@ -509,7 +509,9 @@ protected void DrawTextInternal(DrawingHandleScreen handle, string text, UIBox2
 
         protected override Vector2 CalculateMinimumSize()
         {
-            return (ActualBackground?.MinimumSize.X ?? 0, _itemListHeight);
+            if (_itemListHeight > ActualBackground?.MinimumSize.Y)
```
Comment (PJB3005) [inline Robust.Client/UserInterface/Controls/ItemList.cs]: Wouldn't you just.. need to do ```cs
return (ActualBackground?.MinimumSize.X ?? 0, _itemListHeight + ActualBackground?.MinimumSize.Y ?? 0);
```

Actually I guess if it weren't obvious, Godot uses `MinimumSize` on styleboxes as the "extra width you need to add" which perhaps is *kinda* annoying and should be fixed, but oh well.

Comment (gradientvera) [reply Robust.Client/UserInterface/Controls/ItemList.cs]: Oh, I didn't know about that... Sorry, should be fixed now.

**Review by PJB3005 — APPROVED** [review]

**Pull request merged**